src/dex-ui/store/poolsSlice/poolsSlice.ts

```diff
@@ -236,10 +244,23 @@ const createPoolsSlice: PoolsSlice = (set, get): PoolsStore => {
       set(
         ({ pools }) => {
           pools.withdrawState = initialPoolsStore.withdrawState;
+          pools.poolsPageState = initialPoolsStore.poolsPageState;
           pools.errorMessage = "";
         },
         false,
-        PoolsActionType.RESET_WITHDRAW_STATE
+        PoolsActionType.RESET_WITHDRAW_AND_POOLS_PAGE_STATE
+      );
+    },
+    navigateToPoolsPage: async (tabIndex?: number) => {
+      set(
+        ({ pools }) => {
+          pools.poolsPageState = {
```

Comment:
We're using immer with Zustand so you can immutably update store data with syntax that would typically mutate an object:
```
      set(
        ({ pools }) => {
           pools.poolsPageState.selectedPoolsTabIndex = tabIndex ? tabIndex : 0
         },
        false,
        PoolsActionType.NAVIGATE_TO_POOLS_PAGE
      );
```